### PR TITLE
optimize: use curator instead of zkclient in registry model

### LINKED ISSUE #6772: 当使用jdk21的时候，seata无法连接到zookeeper
### Ⅰ. Issue Description
当使用jdk21的时候，seata无法连接到zookeeper

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!

```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

Minimal yet complete reproducer code (or URL to code):

### Ⅵ. Environment:

- JDK version(e.g. `java -version`): 21
- Seata client/server version:  2.0.0
- Database version:  pg
- OS(e.g. `uname -a`): windows
- Others:

### PR DESCRIPTION
**Pull request by leizhiyuan**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did

fixes https://github.com/apache/incubator-seata/issues/6772

### Ⅲ. Why don't you add test cases (unit test/integration test)? 

there has been

### Ⅳ. Describe how to verify it

ZookeeperRegisterServiceImplTest

### Ⅴ. Special notes for reviews

merge after https://github.com/apache/incubator-seata/pull/6779

### CODE REVIEW

**Comment by funky-eyes** [comment]
ZkConfigurationTest.testRemoveConfig:119 expected: <true> but was: <false>
这个失败率非常高，得看下如何解决，不然合进去时不时就失败不太好

**Comment by leizhiyuan** [comment]
> ZkConfigurationTest.testRemoveConfig:119 expected: but was: 这个失败率非常高，得看下如何解决，不然合进去时不时就失败不太好

image
 
最新的一次应该修复了

**Comment by funky-eyes** [comment]
> > ZkConfigurationTest.testRemoveConfig:119 expected: but was: 这个失败率非常高，得看下如何解决，不然合进去时不时就失败不太好
> 
> image
> 最新的一次应该修复了

ok

File: discovery/seata-discovery-zk/src/main/java/org/apache/seata/discovery/registry/zk/ZookeeperRegisterServiceImpl.java [hunk 1]
```diff
@@ -16,7 +16,10 @@
  */
 package org.apache.seata.discovery.registry.zk;
 
+import com.google.common.annotations.VisibleForTesting;
```
Comment (funky-eyes) [inline discovery/seata-discovery-zk/src/main/java/org/apache/seata/discovery/registry/zk/ZookeeperRegisterServiceImpl.java]:
```suggestion
```

File: discovery/seata-discovery-zk/src/main/java/org/apache/seata/discovery/registry/zk/ZookeeperRegisterServiceImpl.java [hunk 2]
```diff
@@ -28,30 +31,36 @@
...
 import org.apache.seata.config.Configuration;
 import org.apache.seata.config.ConfigurationFactory;
 import org.apache.seata.config.exception.ConfigNotFoundException;
 import org.apache.seata.discovery.registry.RegistryService;
-import org.I0Itec.zkclient.IZkChildListener;
-import org.I0Itec.zkclient.IZkStateListener;
-import org.I0Itec.zkclient.ZkClient;
-import org.apache.zookeeper.Watcher;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
```
Comment (funky-eyes) [inline discovery/seata-discovery-zk/src/main/java/org/apache/seata/discovery/registry/zk/ZookeeperRegisterServiceImpl.java:51]:
```suggestion
import org.apache.zookeeper.KeeperException;

import com.google.common.annotations.VisibleForTesting;
```

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**